### PR TITLE
Cleanup Code for Open Images and others

### PR DESCRIPTION
**Pull request by NikhilBartwal**
* Dataset Name: Open Images, Sun, iNaturalist
* Issue Reference: #2460 
Datasets are generating properly!

### CODE REVIEW

File: tensorflow_datasets/image_classification/inaturalist.py [hunk 1]
```diff
@@ -78,16 +78,11 @@ def _info(self):
   def _split_generators(self, dl_manager):
     output_files = dl_manager.download_and_extract({
         "trainval_images":
-            tfds.download.Resource(
-                url=urllib.parse.urljoin(_URL, "train_val_images.tar.gz"),
-                extract_method=tfds.download.ExtractMethod.NO_EXTRACT),
```
Comment (Conchylicultor) [inline tensorflow_datasets/image_classification/inaturalist.py:83]: Are you sure removing `tfds.download.Resource` is working ? I would expect that the `.tar.gz` is no extracted, so `dl_manager.iter_archive` would crash.

**Review by Conchylicultor — CHANGES_REQUESTED** [review]

File: tensorflow_datasets/image_classification/inaturalist.py (hunk 1 above)
Comment (NikhilBartwal) [reply tensorflow_datasets/image_classification/inaturalist.py:83]: @Conchylicultor I did notice that later on. However, the dataset gives a `download error` using the modified code as well as the original code as follows:

`tensorflow_datasets.core.download.downloader.DownloadError: Failed to get url http://people.vision.caltech.edu/~gvanhorn/datasets/inaturalist/fgvc4_competition/test2017.tar.gz. HTTP code: 403.`

Trying to download the dataset manually using the link in the code gives a permission error

`You don't have permission to access /~gvanhorn/datasets/inaturalist/fgvc4_competition/train_val_images.tar.gz on this server.`

I think we would need to find some other alternate place to get the dataset from

Comment (Conchylicultor) [reply tensorflow_datasets/image_classification/inaturalist.py:83]: It means the website has been updated. And the dataset should be updated (but in another PR). We should not merge changes which are not working

Comment (Keshav15) [reply tensorflow_datasets/image_classification/inaturalist.py:83]: @Conchylicultor  when we try to visit 
http://people.vision.caltech.edu/~gvanhorn/datasets/inaturalist/fgvc4_competition/
then we have 3 files there but not (train_val_images.tar.gz)
so exactly what should be done?

Comment (NikhilBartwal) [reply tensorflow_datasets/image_classification/inaturalist.py:83]: @Conchylicultor Yes, I think we should leave the `inaturalist` dataset and merge the other two

Comment (Conchylicultor) [reply tensorflow_datasets/image_classification/inaturalist.py:83]: It's depends, are those datasets working  with those changes ?

**Review by Conchylicultor — CHANGES_REQUESTED** [review]

**Comment by NikhilBartwal** [comment]
@Conchylicultor The changes in the Open images dataset are compatible with the original code whereas, the `paths` variable in the Sun dataset was best used using the original `tfds.download.resource` method, which has been fixed now. I thinkl the `Open Images` and the `Sun` dataset PRs can be merged now. The `inaturalist` is best with the original code until we find another website to get the dataset from

**Comment by vijayphoenix** [comment]
Closing the PR for now https://github.com/tensorflow/datasets/pull/2479#discussion_r498882091

**Pull request closed without merging**